File: src/jcleasing/models/units.py
"""Data models for apartment units and pricing information."""
from dataclasses import dataclass
from typing import List


@dataclass
class PriceInfo:
    """Represents pricing information for a unit."""
    price: int = 0
    price_range: str = ""
    date_fetched: str = ""


@dataclass
class UnitInfo:
    """Represents information about an apartment unit."""
    unit: str = ""
    building: str = ""
    size: int = 0
    available_date: str = ""
    floorplan_type: str = ""
    floorplan_link: str = ""
    floorplan_note: str = ""
    prices: List[PriceInfo] = None
    
    def __post_init__(self):
        if self.prices is None:
            self.prices = []

File: data_display/src/types/units.ts
export interface PriceInfo {
  price: string | number
  price_range: string
  date_fetched: string
}

export interface UnitInfo {
  unit: string
  building: string
  size: string | number
  available_date: string
  floorplan_type: string
  floorplan_link: string
  floorplan_note: string
  prices: PriceInfo[]
}

export interface FilterOptions {
  building: string[]
  floorplanType: string[]
  minPrice: number | null
  maxPrice: number | null
  availableFrom: string | null
  availableTo: string | null
  searchQuery: string
}

export interface SortOptions {
  key: string
  order: 'asc' | 'desc'
}

export interface UnitStoreState {
  units: UnitInfo[]
  loading: boolean
  error: string | null
  filters: FilterOptions
  sort: SortOptions
}
